pkg/pmem-csi-driver/pmem-csi-driver.go

```diff
@@ -230,12 +226,51 @@ func (pmemd *pmemDriver) registerNodeController() error {
 		time.Sleep(retryTimeout)
 	}
 
-	client := registry.NewRegistryClient(conn)
+	req := &registry.RegisterControllerRequest{
+		NodeId:   pmemd.cfg.NodeID,
+		Endpoint: pmemd.cfg.ControllerEndpoint,
+	}
+
+	if err := register(context.Background(), conn, req); err != nil {
+		return err
+	}
+	go waitAndWatchConnection(conn, req)
+
+	return nil
+}
+
+// waitAndWatchConnection Keeps watching for connection changes, and whenever the
+// connection state changed from lost to ready, it re-register the node controller with registry server.
+func waitAndWatchConnection(conn *grpc.ClientConn, req *registry.RegisterControllerRequest) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+
+	connectionLost := false
+
+	for {
+		s := conn.GetState()
+		if s == connectivity.Ready {
+			if connectionLost {
+				glog.Info("ReConnected.")
```

Comment:
nit: ReConnected -> Reconnected

Reply:
> I fix CLEAR version to 29820 per knowledge this is the working setup ATM (is this correct?)

I was testing it on 29880, but it shouldn't make much difference I guess.